src/status_im/browser/multi_chain.cljs

```diff
@@ -0,0 +1,160 @@
+(ns status-im.browser.multi-chain
+  (:require [status-im.constants :as constants]
+            [clojure.string :as string]
+            [status-im.utils.debounce :as debounce]
+            [re-frame.core :as re-frame]
+            [status-im.utils.fx :as fx]
+            [status-im.ethereum.core :as ethereum]
+            [status-im.network.core :as network]
+            [taoensso.timbre :as log]
+            [status-im.utils.random :as random]
+            [status-im.ethereum.json-rpc :as json-rpc]
+            [status-im.ui.screens.browser.multi-chain.sheet :as sheet]))
+
+(defn- wallet-connected? []
+  @(re-frame/subscribe [:browser/wallet-connected?]))
+
+(fx/defn send-success-call-to-bridge
+  {:events [:multi-chain/send-success-call-to-bridge]}
+  [_ id messageId]
+  {:browser/send-to-bridge {:type constants/web3-send-async-callback
+                            :messageId messageId
+                            :result {:jsonrpc "2.0"
+                                     :id (int id)
+                                     :result nil}}})
+
+(fx/defn send-permission-reject-error-to-bridge
+  {:events [:multi-chain/send-permission-reject-error-to-bridge]}
+  [_ message-id]
+  {:browser/send-to-bridge {:type constants/web3-send-async-callback
+                            :messageId message-id
+                            :error {:code 4001
+                                    :message "User rejected the request."}}})
+
+(defn switch-ethereum-chain? [type]
+  (= type constants/multi-chain-switch-ethereum-chain))
+
+(defn add-ethereum-chain? [type]
+  (= type constants/multi-chain-add-ethereum-chain))
+
+(defn get-switch-ethereum-chain-permission [params]
+  {:type :switch-ethereum-chain
```

Comment:
im not sure this still need this, we could move it to ui and drop type , just use `method` param